src/specific/s_audio_stream.c

```diff
@@ -390,6 +394,7 @@ bool S_Audio_StreamSoundClose(int sound_id)
     stream->is_playing = false;
     stream->is_looped = false;
     stream->volume = 0.0f;
+    stream->timestamp = 0;
```

Comment:
There's lots of similar code here,  I can see opportunity to extract it to a common procedure

Reply:
I was going to make this function for `S_Audio_StreamSoundInit` and for `S_Audio_StreamSoundClose`.
```
static void S_Audio_StreamSoundClear(AUDIO_STREAM_SOUND *stream)
{
    stream->is_used = false;
    stream->is_playing = false;
    stream->is_looped = false;
    stream->is_read_done = true;
    stream->volume = 0.0f;
    stream->timestamp = 0;
    stream->sdl.stream = NULL;
    stream->finish_callback = NULL;
}
```

But I'm unsure what this part is doing in `S_Audio_StreamSoundClose`:
```
    if (stream->finish_callback) {
        stream->finish_callback(sound_id, stream->finish_callback_user_data);
    }
```
Because `S_Audio_StreamSoundInit` sets this to: `stream->finish_callback = NULL;`

Reply:
The suggestion above is OK, however, the clear routine has to happen after the stream finish callback gets executed. It's used to play looped background tracks.

Reply:
Ok lmk if what I pushed works for you.